cluster-autoscaler/expander/grpcplugin/grpc_client.go

```diff
@@ -0,0 +1,129 @@
+package grpcplugin
+
+import (
+  "context"
+  "log"
+  "time"
+
+  v1 "k8s.io/api/core/v1"
+  "k8s.io/autoscaler/cluster-autoscaler/expander"
+  "k8s.io/autoscaler/cluster-autoscaler/expander/grpcplugin/protos"
+  "k8s.io/klog/v2"
+  schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+
+  "google.golang.org/grpc"
+  "google.golang.org/grpc/credentials"
+)
+
+type grpcclientstrategy struct {
+  grpcClient protos.ExpanderClient
+}
+
+type GRPCOpts struct {
+  GRPCExpanderCert string
+  GRPCExpanderUrl string
+}
+
+// NewFilter returns an expansion filter that creates a gRPC client, and calls out to a gRPC server
+func NewFilter(opts GRPCOpts) expander.Filter {
+  client := createGRPCClient(opts)
+  if client == nil {
```

Comment:
I'm not sure I follow here... if `client == nil`, doesn't that indicate an error?  Seems like it would be better to have `createGRPCClient` return `protos.ExpanderClient, error`?

Reply:
Discussed on slack, this seems fine.